Add all of the no-op segmentation mask augmentations

ianstenbit opened the pull request:
This adds "segmentation map support" to all the layers that shouldn't change a segmentation map (mostly because they're just color adjustments)

fchollet reviewed (commented):
Thanks for the PR!

For features like this, we want to make sure that we don't forget to implement support for the feature in new layers that we ship. This requires having a unit test that automatically checks the presence of the feature (similar to `get_config()` checks in the standard layer test).

qlzh727 commented:
Agreed. We probably want to have a test like https://github.com/keras-team/keras-cv/blob/master/keras_cv/layers/preprocessing/with_labels_test.py for segmentation_map. Basically a layer_test for all the common functionality of image KPLs.

qlzh727 reviewed (changes requested):
Please add a test case for common no-op layers.

qlzh727 commented on keras_cv/layers/preprocessing/with_segmentation_masks_test.py:
```diff
@@ -0,0 +1,101 @@
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.float32
+        )
+        segmentation_masks = tf.random.uniform(
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.int32
```
The shape of the mask should probably be 3, 512, 512, 1

ianstenbit replied:
Done

ianstenbit commented:
I've added a with_segmentation_mask test, but I'm not sure that this fully solves the problem -- we still need to manually make sure these tests include all of the relevant layers (which the with_labels test currently did not).

These tests also don't assert that the segmentation mask and/or label are actually passed through in the output, so I've updated the with_labels test to do that and done the same in with with_segmentation_mask test.

qlzh727 commented on keras_cv/layers/preprocessing/with_segmentation_masks_test.py:
```diff
@@ -0,0 +1,101 @@
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.float32
+        )
+        segmentation_masks = tf.random.uniform(
+            shape=(3, 512, 512, 3), minval=0, maxval=1, dtype=tf.int32
```
The value range should be 0 ~ classes, and 1 seems to be a bad choice here.

ianstenbit replied:
Done

qlzh727 commented on keras_cv/layers/preprocessing/with_segmentation_masks_test.py:
```diff
@@ -0,0 +1,101 @@
+        )
+
+        inputs = {"images": img, "segmentation_masks": segmentation_masks}
+        _ = layer(inputs)
```
We should validate that the output mask is not changed (no op).

ianstenbit replied:
Done, although this won't always be true (once we add non-noop KPLs for seg masks). I've left a comment about this, and we can add a parameter for this to the tests in the future.

qlzh727 commented on keras_cv/layers/preprocessing/with_segmentation_masks_test.py:
```diff
@@ -0,0 +1,101 @@
+            shape=(512, 512, 3), minval=0, maxval=1, dtype=tf.float32
+        )
+        segmentation_mask = tf.random.uniform(
+            shape=(512, 512, 3), minval=0, maxval=1, dtype=tf.int32
```
same here.

ianstenbit commented:
/gcbrun

qlzh727 commented:
I think we probably  

> I've added a with_segmentation_mask test, but I'm not sure that this fully solves the problem -- we still need to manually make sure these tests include all of the relevant layers (which the with_labels test currently did not).
> 
> These tests also don't assert that the segmentation mask and/or label are actually passed through in the output, so I've updated the with_labels test to do that and done the same in with with_segmentation_mask test.

qlzh727 commented on keras_cv/layers/preprocessing/base_image_augmentation_layer.py:
```diff
@@ -28,7 +28,7 @@
 KEYPOINTS = "keypoints"
 RAGGED_BOUNDING_BOXES = "ragged_bounding_boxes"
-SEGMENTATION_MASK = "segmentation_mask"
+SEGMENTATION_MASK = "segmentation_masks"
```
SEGMENTATION_MASK[S]?

ianstenbit replied:
Done. Made it plural for consistency with the other labels

ianstenbit commented:
/gcbrun

ianstenbit commented:
/gcbrun

LukeWood reviewed (approved):
Looks good to me, thanks Ian!
